notifications_api_common/tasks.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+import celery
+import requests
+from zds_client import ClientError
+
+from .models import NotificationsConfig
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigTask(celery.Task):
+    """
+    add retry options from NotificationsConfig into the task instance
+    """
+
+    def __init__(self):
+        config = NotificationsConfig.get_solo()
+
+        self.max_retries = config.notification_delivery_max_retries
+        self.retry_backoff = config.notification_delivery_retry_backoff
+        self.retry_backoff_max = config.notification_delivery_retry_backoff_max
+
+
+class NotificationException(Exception):
+    pass
+
+
+@celery.current_app.task(
+    bind=True,
+    autoretry_for=(NotificationException, requests.RequestException),
+    base=ConfigTask,
+)
```

Comment:
is there a reason you're not using [`@shared_task`](https://docs.celeryq.dev/en/stable/django/first-steps-with-django.html#using-the-shared-task-decorator) for this instead?